patch/llpcConfigBuilderBase.cpp

```diff
@@ -376,13 +376,22 @@ void ConfigBuilderBase::WritePalMetadata()
 
     // Add the register values to the MsgPack document.
     msgpack::MapDocNode registers = m_pipelineNode[".registers"].getMap(true);
-    size_t sizeInDword = m_configSize / sizeof(uint32_t);
+    const size_t sizeInDwords = m_configSize / sizeof(uint32_t);
     // Configs are composed of DWORD key/value pair, the size should be even
-    LLPC_ASSERT((sizeInDword % 2) == 0);
-    for (size_t i = 0; i < sizeInDword; i += 2)
+    LLPC_ASSERT((sizeInDwords % 2) == 0);
+    static_assert(sizeof(*m_pConfig) == 1, "Expected a byte buffer");
+    const uint8_t* pCurrentConfig = m_pConfig;
+    const uint8_t* const pConfigEnd = m_pConfig + m_configSize;
```

Comment:
Is `m_pConfig` a byte buffer or a word buffer?
Changing the pointer type here is a very confusing thing to me for the following reasons:
- It raises questions about legality of the code -- reading memory as a different type than the last written can exhibits undefined behavior in plain C++, and is legal in certain scenarios. In the case of LLPC, it seems legal only because it is compiled with `-fno-strict-aliasing`, but there is no way of telling that without analyzing the build scripts.
- It raises doubts about other low-level issues such as alignment, type sizes, etc.

I think that making type punning more explicit here solves the above issues and reduces mental overhead required to understand it convince yourself that the code is doing what was it was intended to.

Reply:
m_pConfig is a word buffer. but we need change multiple files/functions when change the type from uint8_t* to uint32_t*